nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -174,16 +178,36 @@ def __init__(self, regressor: str, data_dimension: int):
             assert False, f"Problem type {regressor} undefined!"
         self.register_initialization(regressor=regressor, data_dimension=data_dimension)
 
-    def get_dataset(self, data_dimension):
+    def get_dataset(self, data_dimension, dataset_name):
+
+        if dataset_name != "artificial":
+            assert dataset_name in ["boston", "diabetes"]
+            data = {"boston": sklearn.datasets.load_boston,
+                    "diabetes": sklearn.datasets.load_diabetes,
+                    }[dataset_name](return_X_y=True)
+            # Half the dataset for training.
+            self.X = data[0][::2]
+            self.y = data[1][::2]
+            num_train_data = len(self.X)
+            self.num_data = num_train_data
+            for cv in range(10):
+                train_range = np.arange(num_train_data) % 10 != cv 
+                valid_range = np.arange(num_train_data) % 10 == cv
+                self.X_train += [self.X[train_range]]
+                self.y_train += [self.y[train_range]]
+                self.X_valid += [self.X[valid_range]]
+                self.y_valid += [self.y[valid_range]]
+            self.X_test = data[0][1::2]
+            self.y_test = data[1][1::2]
```

Comment:
all the attribues need to be typed in the init

Reply:
only num_data is missing; I add it.